core/schema/module.go

```diff
@@ -280,6 +283,18 @@ func (s *moduleSchema) Install() {
 
 		dagql.Func("withConstructor", s.typeDefWithObjectConstructor).
 			Doc(`Adds a function for constructing a new instance of an Object TypeDef, failing if the type is not an object.`),
+
+		dagql.Func("withEnum", s.typeDefWithEnum).
+			Doc(`Returns a TypeDef of kind Enum with the provided name.`,
+				`Note that an enum's values may be omitted if the intent is only to refer to an enum.
+				This is how functions are able to return their own, or any other circular reference.`).
+			ArgDoc("name", `The name of the enum`).
+			ArgDoc("description", `A doc string for the enum, if any`),
+
+		dagql.Func("withEnumValue", s.typeDefWithEnumValue).
+			Doc(`Adds a static value for an Enum TypeDef, failing if the type is not an enum.`).
+			ArgDoc("value", `The name of the value in the enum`).
+			ArgDoc("description", `A doc string for the value, if any`),
```

Comment:
Hm, why do we have a `withEnumValue`? Couldn't `withEnum` instead accept an array? At point of instantiation we should know all the possible values right?

Not a deal breaker at all, I think I have a preference for having just one method.

Reply:
Each value also has a description, so I can see the reasoning here. To pass them all in as an array you'd need to pass a structure that has both, right?

Reply:
Mm, you're right. Yup, I guess in that case, we should keep this as is.

Reply:
I just followed the API @helderco designed, I'm not sure which design is the best though